x-pack/plugins/enterprise_search/public/applications/shared/api_logic/create_api_logic.ts

```diff
@@ -23,19 +24,46 @@ export interface Actions<Args, Result> {
   makeRequest(args: Args): Args;
 }
 
+export interface CreateApiOptions<Result> {
+  clearFlashMessagesOnMakeRequest: boolean;
+  showErrorFlash: boolean;
+  showSuccessFlashFn?: (result: Result) => string;
+}
+
+const DEFAULT_CREATE_API_OPTIONS = {
+  clearFlashMessagesOnMakeRequest: true,
```

Comment:
I'm a little torn on defaulting this to `true`. It does give me pause and worries if we're using > 1 api logic they could potentially step on each other, especially in the case of polling or re-fetching.

But I _think_ overall I am in favor this.

Reply:
Yes, I think we should be a little more attentive to odd behavior around this logic the next couple of weeks. I'm sure I missed a logic or edge case here or there, but adding a flag to set this top false to a logic is pretty easy.